cdc/cdc/processor/pipeline/sorter.go

```diff
@@ -151,6 +151,30 @@ func (n *sorterNode) StartActorNode(ctx pipeline.NodeContext, eg *errgroup.Group
 						}
 					}
 
+					// We calculate memory consumption by PolymorphicEvent size.
+					size := uint64(msg.RawKV.ApproximateDataSize())
+					// NOTE we allow the quota to be exceeded if blocking means interrupting a transaction.
+					// Otherwise the pipeline would deadlock.
+					err := n.flowController.Consume(commitTs, size, func() error {
+						if lastCRTs > lastSentResolvedTs {
+							// If we are blocking, we send a Resolved Event here to elicit a sink-flush.
+							// Not sending a Resolved Event here will very likely deadlock the pipeline.
+							lastSentResolvedTs = lastCRTs
+							lastSendResolvedTsTime = time.Now()
+							ctx.SendToNextNode(pipeline.PolymorphicEventMessage(model.NewResolvedPolymorphicEvent(0, lastCRTs, n.keyspanID)))
+						}
+						return nil
+					})
+					if err != nil {
+						if cerror.ErrFlowControllerAborted.Equal(err) {
+							log.Info("flow control cancelled for keyspan",
+								zap.Uint64("keyspanID", n.keyspanID),
+								zap.String("keyspanName", n.keyspanName))
+						} else {
+							ctx.Throw(err)
```

Comment:
Does this cause panic?

Reply:
It does not cause panic, just throws a err to parents ndoes.